puppet/spec/unit/indirector/catalog/puppetdb_spec.rb

```diff
@@ -446,6 +446,28 @@ def catalog_data_hash
           result['edges'].should include(edge)
         end
 
+        it "should add edges which refer to collected virtual resources with hyphens in the classname" do
+          Puppet[:code] = <<-MANIFEST
+          define foo-bar- (){}
+          @foo-bar- { 'baz': }
+
+          notify { source:
+            before => Foo-bar-[baz],
+          }
+
+          Foo-bar- <| |>
+          MANIFEST
+
+          result = subject.munge_catalog(catalog)
+          other_edge = {
+            'source' => {'type' => 'Notify', 'title' => 'source'},
+            'target' => {'type' => 'Foo-bar-', 'title' => 'baz'},
+            'relationship' => 'before'
+          }
+
+          result['edges'].should include(other_edge)
+        end
+
```

Comment:
@kbarber so I wasn't sure if this spec tests goes through all the validation we were talking about making an acceptance test for(?)